src/ptvsd/common/attach_script_ptvsd_pid.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import os.path
+
+
+__file__ = os.path.abspath(__file__)
+_ptvsd_dir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+
+
+def attach(port, host, client, log_dir):
+    try:
+        import sys
+        if 'threading' not in sys.modules:
+            try:
+
+                def on_warn(msg):
+                    from ptvsd.commom import log
+                    log.warn(msg)
+
+                def on_exception(msg):
+                    from ptvsd.common import log
+                    log.exception(msg)
+
+                def on_critical(msg):
+                    from ptvsd.common import log
+                    log.error(msg)
```

Comment:
`ptvsd` is not loaded here because we need to do the `sys.path` dance first. If these get called by `fix_main_thread_id()`, it would also be the case, so I don't think trying to load it deferred is going to help. At this point, the best you can do is print to stderr.